tests/test_tls/test_tls.c

```diff
@@ -67,8 +67,13 @@ int solo5_app_main(const struct solo5_start_info *si __attribute__((unused)))
 {
     puts("\n**** Solo5 standalone test_tls ****\n\n");
 
+#if defined (__powerpc64__)
+    tcb1.tp = (void *)&tcb1._data + 0x7000;
+    tcb2.tp = (void *)&tcb2._data + 0x7000;
```

Comment:
Thanks for changing this (moving the magic constant out of solo5_set_tls_base() itself). Additionally, in the test itself, can we #define this value as, say, `PPC64_TLS_OFFSET` and add a comment pointing the reader to Figure 3.1 here http://openpowerfoundation.org/wp-content/uploads/resources/leabi/content/dbdoclet.50655241_50444.html#dbdoclet.50655241_11666?

Reply:
I'll add the #define. Shouldn't it be in a public header file so other higher layers can reuse it? I also think that no higher layer (like test app) should be aware of the offset but just tell the API which TCB to use and the adjustment would happen where the register is loaded with the address...

Reply:
Just add it in the test case for now, this is not part of the Solo5 API. Thanks,

Reply:
Added and pushed the changes.

Reply:
What about this new `struct tcb` for ppc to not have the special case at line 70?

```
struct tcb {
     volatile uint64_t _data;
     char pad[PPC64_TLS_OFFSET - 8]
     void *tp;
};
```

Reply:
I had something like this before...

Reply:
That ppc64 adjusts the pointer is really just an artifact of efficient addressing modes the processor supports and should/could be hidden. The application/library just defines what the layout of the TCB should be, e.g. the order of the thread-local variables, and with this the TCB's size and where it should be (address).

Reply:
... I'm still slightly confused about when the magic 0x7000 gets used. What is the disassembly of `set_data()` and `get_data()` when compiled for PPC?

Reply:
```
#  objdump -D tests/test_tls/test_tls.spt | less
[...]
0000000000103670 <get_data>:
  103670:       00 00 00 60     nop
  103674:       00 90 2d 39     addi    r9,r13,-28672
  103678:       00 00 69 e8     ld      r3,0(r9)
  10367c:       20 00 80 4e     blr
        ...
  10368c:       00 00 42 60     ori     r2,r2,0

0000000000103690 <set_data>:
  103690:       00 00 00 60     nop
  103694:       00 90 2d 39     addi    r9,r13,-28672
  103698:       00 00 69 f8     std     r3,0(r9)
  10369c:       20 00 80 4e     blr
        ...
  1036ac:       00 00 42 60     ori     r2,r2,0
````

Reply:
So, it subtracts 0x7000 from the contents of register r13 to get to the 'data' variable.